### DISCUSSION TITLE
JSX preserve

### DISCUSSION THREAD

**Discussion by pnemi** [vercel/next.js#11216] [Help]
Hello,

why does Next requires tsconfig option "jsx" set to "preserve"?
Does TypeScript not transpiles JSX and passes it to (some internal) Babel etc. or is is entirely different reason?

:)

**Comment by leeren** [comment]
I'd also like to know why? Is this needed?

**Reply by techmind-git** [reply]
I'd like to know as well since I need it to stay "react-jsx" to be able to compile mdx-bundler imported components with eslint. Eslint won't work otherwise (https://github.com/evanw/esbuild/issues/2634). nextjs keeps changing my tsconfig setting to 'preserve'.
